server/routes/experiments/biomed_nl/api.py

```diff
@@ -80,14 +89,29 @@ def _append_fallback_response(query, response, path_finder,
       SELECTED_PATHS=utils.format_dict(selected_paths),
       ENTITY_INFO=utils.format_dict(entity_info))
 
+  token_count = gemini_client.models.count_tokens(
+      model=GEMINI_PRO,
+      contents=fallback_prompt,
+  ).total_tokens
+  while token_count > GEMINI_PRO_TOKEN_LIMIT:
+    logging.info('[biomed_nl] truncate fallback prompt')
```

Comment:
nit: instead of logging this for every round of the while loop, can we just do it once? maybe get the count before and after the while loop & log if there is a difference

Reply:
the loop should really only run one time; if it's running more than that, then I want to know so that I can fix it. I updated to log total number of truncation loops instead of each loop as a warning if it ran more than once.